backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -309,16 +264,41 @@ private Project insertProject(Project project, String workspaceName, UserInfo us
     }
   }
 
+  private void removeDeletedProjectByNameFromUAC(String projectName, Session session) {
+    Query deletedEntitiesQuery = session.createQuery(GET_DELETED_PROJECTS_IDS_BY_NAME_HQL);
+    deletedEntitiesQuery.setParameter("projectName", projectName);
+    List<String> deletedEntityIds = deletedEntitiesQuery.list();
+    if (!deletedEntityIds.isEmpty()) {
+      try {
+        CommonUtils.registeredBackgroundUtilsCount();
+        roleService.deleteEntityResourcesWithServiceUser(
+            deletedEntityIds, ModelDBServiceResourceTypes.PROJECT);
+      } finally {
+        CommonUtils.unregisteredBackgroundUtilsCount();
+      }
+    }
+  }
+
   @Override
   public Project updateProjectName(UserInfo userInfo, String projectId, String projectName)
       throws InvalidProtocolBufferException {
     try (Session session = modelDBHibernateUtil.getSessionFactory().openSession()) {
-      // TODO: Remove this after UAC support update entity name using SetResource
       Workspace workspace = roleService.getWorkspaceByWorkspaceName(userInfo, null);
-      checkIfEntityAlreadyExists(session, workspace, projectName);
+      removeDeletedProjectByNameFromUAC(projectName, session);
 
       ProjectEntity projectEntity =
           session.load(ProjectEntity.class, projectId, LockMode.PESSIMISTIC_WRITE);
+      Project project = projectEntity.getProtoObject(roleService, authService);
+      roleService.createWorkspacePermissions(
+          workspace.getId(),
+          Optional.empty(),
+          projectId,
+          projectName,
+          Optional.empty(), // UAC will populate the owner ID
```

Comment:
This means we might end up changing the owner! Let's not do this as it's a bug. First you get, then update the fields, then save the updated object. Please make sure you _only_ change the field you wanted to change.

Reply:
Yes i am sure i am updating only the two fields.
```
projectEntity.setName(projectName);
projectEntity.setDate_updated(Calendar.getInstance().getTimeInMillis());
```

Reply:
`Optional.empty(), // UAC will populate the owner ID` was changed at first phase of visibility changes where we have moved owner field to UAC so we have used `Optional.empty()` everywhere. we will remove this unused parameter later.

Reply:
You're probably assuming I know the inner workings on `roleService.createWorkspacePermissions`, which I do not and the name `create` indicates to me that it might behave wrong on an `update`. I am basing on the API. If you call the `setResources` API with **any** field different from the value you get on a `getResources` API (including empty), you'll do an update. Are you _absolutely_ sure this will _never_ happen in that method? If yes, let's fix the comment in this line because it's describing the wrong behavior.

Reply:
Removed `updateProjectName` method and additionally added java doc comment on the method `roleService.createWorkspacePermissions` to understand the method parameters   https://github.com/VertaAI/modeldb/pull/1946/files